sdk/python/tests/unit/infra/scaffolding/test_repo_config.py

```diff
@@ -224,3 +225,71 @@ def test_invalid_project_name():
         ),
         expect_error="alphanumerical values ",
     )
+
+
+def test_auth_config():
+    allowed_auth_types = [authType.value for authType in AuthType]
+    _test_config(
+        dedent(
+            """
+        project: foo
+        auth:
+            client_id: test_client_id
+            client_secret: test_client_secret
+            username: test_user_name
+            password: test_password
+            realm: master
+            auth_server_url: http://localhost:8712
+        registry: "registry.db"
+        provider: local
+        online_store:
+            path: foo
+        entity_key_serialization_version: 2
+        """
+        ),
+        expect_error=f"auth configuration is not having authentication type. Possible values={allowed_auth_types}",
```

Comment:
the test function just validate that `expect_error in str(error)`, I think you can restrict the `expect_error` message to the bare minimum to avoid reviewing the test code in case the actual message changes